### PR TITLE
inference pipelines using clarify

### PR DESCRIPTION
**Pull request by vikeshpandey**
*Issue #, if available:*
NA

*Description of changes:*

This example shows how to use SageMaker Clarify using sagemaker inference pipelines.

*Testing done:*
Yes

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [ x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [ x] I have tested my notebook(s) and ensured it runs end-to-end
- [x ] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sagemaker-clarify/clarify-explainability-inference-pipelines/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## Credit risk prediction and explainability with Amazon SageMaker
+
+This example shows how to user SageMaker Clarify to run explainability jobs on a SageMaker hosted inference pipelines.
```
Comment (jkroll-aws) [inline sagemaker-clarify/clarify-explainability-inference-pipelines/README.md:3]: Grammar nit - `a SageMaker hosted inference pipeline` or `SageMaker hosted inference pipelines`

File: sagemaker-clarify/clarify-explainability-inference-pipelines/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Below is the architecture diagram used in the solution:
+
+![alt text](clarify_inf_pipeline_arch.png)
+
+
+The notebook performs the following steps:
+
+1. Prepare raw training and test data
+2. Create a SageMakerProcessing job which performs preprocessing on the raw training data and also produce an SKlearn model which is reused while deployment.
```
Comment (jkroll-aws) [inline sagemaker-clarify/clarify-explainability-inference-pipelines/README.md:13]: Grammar revisions - `Create a SageMaker Processing job which performs preprocessing on the raw training data and also produces an SKlearn model which is reused for deployment.`

File: sagemaker-clarify/clarify-explainability-inference-pipelines/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+3. Train an XGBoost model on the processed data using SageMaker's built-in XGBoost container.
+4. Create a SageMaker Inference pipeline containing the SKlearn and XGBoost model in a series.
+5. Perform inference by supplying raw test data
+6. Set up and run explainability job powered by SageMaker Clarify
+7. Use open source shap library to create summary and waterfall plots to understand the feature importance better.
+8. Run bias analysis jobs
+9. Clean up
```
Comment (jkroll-aws) [inline sagemaker-clarify/clarify-explainability-inference-pipelines/README.md:20]: Be consistent with using a period or no period at the end of each list item

File: sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb [hunk 4]
```diff
@@ -0,0 +1,1482 @@
+{
```
Comment (jkroll-aws) [inline sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: <p>Edit the first image to remove the red typo squiggles from <code>SageMaker</code>  and <code>explainability</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2866/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (jkroll-aws) [inline sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: <p>Preferred to add code to download from S3, so customers don't need to manually download and upload the dataset. Are you using the SouthGermanCredit.asc file from that link? Please use our hosted version<code>s3://sagemaker-sample-files/datasets/tabular/uci_statlog_german_credit_data/SouthGermanCredit.asc</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/2866/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: sagemaker-clarify/clarify-explainability-inference-pipelines/README.md (hunk 3 above)
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/README.md:20]: fixed

File: sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb (hunk 4 above)
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: fixed

[the remark this answers is not in the document]
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: fixed

[the remark this answers is not in the document]
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: fixed

[the remark this answers is not in the document]
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: fixed

[the remark this answers is not in the document]
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: removed pip install

File: sagemaker-clarify/clarify-explainability-inference-pipelines/README.md (hunk 2 above)
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/README.md]: fixed

File: sagemaker-clarify/clarify-explainability-inference-pipelines/README.md (hunk 1 above)
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/README.md]: fixed

File: sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb (hunk 4 above)
Comment (vikeshpandey) [reply sagemaker-clarify/clarify-explainability-inference-pipelines/credit_risk_explainability_inference_pipelines.ipynb:1]: fixed

**Review by jkroll-aws — APPROVED** [review]

**Pull request merged**